crates/linter/src/plugin/redundancy/rules/redundant_write_visibility.rs

```diff
@@ -0,0 +1,55 @@
+use mago_ast::{Modifier, Property};
+use mago_reporting::{Annotation, Issue, Level};
+use mago_span::HasSpan;
+use mago_walker::Walker;
+
+use crate::{context::LintContext, rule::Rule};
+
+#[derive(Clone, Debug)]
+pub struct RedundantWriteVisibility;
+
+impl Rule for RedundantWriteVisibility {
+    fn get_name(&self) -> &'static str {
+        "redundant-write-visibility"
+    }
+
+    fn get_default_level(&self) -> Option<mago_reporting::Level> {
+        Some(Level::Help)
+    }
+}
+
+impl<'a> Walker<LintContext<'a>> for RedundantWriteVisibility {
+    fn walk_in_property(&self, property: &Property, context: &mut LintContext<'a>) {
+        let modifiers = property.modifiers();
+
+        if modifiers.is_empty() {
+            return;
+        }
+
+        let Some(write_visibility) = modifiers.get_first_write_visibility() else {
+            return;
+        };
+
+        let Some(read_visibility) = modifiers.get_first_read_visibility() else {
+            return;
+        };
+
+        match (read_visibility, write_visibility) {
+            (Modifier::Public(_), Modifier::PublicSet(_)) | (Modifier::Protected(_), Modifier::ProtectedSet(_)) | (Modifier::Private(_), Modifier::PrivateSet(_)) => {
+                let issue = Issue::new(context.level(), "identical write visibility has no effect")
+                    .with_help("remove the redundant write visibility modifier")
+                    .with_annotations(vec![
+                        Annotation::secondary(read_visibility.span()),
+                        Annotation::secondary(write_visibility.span()),
```

Comment:
```suggestion
                        Annotation::primary(write_visibility.span())
                              .with_message("redundant write visibility."),
```

we should highlight this as primary since it is the span with the "problem".